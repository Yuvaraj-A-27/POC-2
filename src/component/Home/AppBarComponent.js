import { AppBar, Button, IconButton, makeStyles, Toolbar, Typography } from '@material-ui/core';
import React from 'react';
import  MenuIcon from '@material-ui/icons/Menu'

const useStyles = makeStyles((theme) => ({
    root: {
      flexGrow: 1,
      marginTop : '-10px',
      top: 0,
    },
    menuButton: {
      marginRight: theme.spacing(16),
    },
    title: {
      flexGrow: 1,
    },
    btn:{
        padding:'0px 10px',
        '&:hover' :{
            backgroundColor:'red',
        }
    }
  }));

function AppBarComponent({id}){
    const classes = useStyles()


    return(
        <div>
            <AppBar id={id} position="sticky" className={classes.root}>
                <Toolbar>
                <IconButton edge="start" className={classes.menuButton} color="inherit" aria-label="menu">
                    <MenuIcon />
                </IconButton>
                <Typography variant="h5" className={classes.title}>
                    MyStore
                </Typography>
                <Button className = {classes.btn} color="inherit">Register</Button>
                <Button className = {classes.btn} color="inherit">Login</Button>
                </Toolbar>
            </AppBar>
        </div>
    )
}

export default AppBarComponent;