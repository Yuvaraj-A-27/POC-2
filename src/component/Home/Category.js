import { makeStyles, Paper, Typography } from '@material-ui/core'
import axios from 'axios'
import React, { useEffect, useState } from 'react'

const useStyles = makeStyles((theme)=>({
    heading:{
        textAlign:'left',
        marginTop:'3%',
        marginLeft : '3%',
        marginRight : '3%',
        fontWeight:'800',
        background: 'linear-gradient(10deg, #9269fa 40%, #d9d9d9 100%)',
        border: 0,
        fontSize: 20,
        borderRadius: 3,
        color: 'white',
        height: 25,
        padding: '10px 30px',


    },
    hr:{
        marginTop:'1%',
        margin: theme.spacing(5)
    },
    paperDiv:{
        display : 'flex',
        flexWrap : 'wrap',
        '& > *': {
            marginLeft: theme.spacing(10),
            marginTop: theme.spacing(1),
            width: theme.spacing(25),
            height: theme.spacing(7),
        }
    },
    paper:{
        background: 'linear-gradient(45deg, #FE6B8B 30%, #FF8E53 90%)',
        fontSize : 28,
        padding: '2px 20px 0px 20px',
        color: 'white',
        height : '10%',
    }
}))

function Category(){
    const classes = useStyles()

    const [category, setCategory] = useState()

    useEffect(()=>{
        axios.get("https://fakestoreapi.com/products/categories")
        .then(res => setCategory(res.data))
        .catch(err =>console.log(err))
    },[])

    const categories = category.map((e,index)=>(
        <Paper className={classes.paper} elevation={3} key={index}>{e}</Paper>
    ))
    return(
        <div>
            <Typography className={classes.heading} variant ='h5'>Category</Typography>
            <hr className={classes.hr}/>
            <div className={classes.paperDiv}>
                {categories}
            </div>
        </div>
    )
}

export default Category