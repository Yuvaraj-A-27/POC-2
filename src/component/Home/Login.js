import { Dialog, DialogActions, DialogContent, DialogTitle, makeStyles, TextField } from '@material-ui/core'
import React, { useState } from 'react'
import Button from '@material-ui/core/Button';
import { connect } from 'react-redux';
import { activeUser, loginActive, loginError, loginErrorTyping } from '../../Store/Action';
import axios from 'axios';
import { useHistory } from 'react-router';

const useStyles = makeStyles((theme)=>({
    margin: {
        margin: theme.spacing(1),
    },

    title:{
        margin: theme.spacing(2)
    },
    content:{
        margin: theme.spacing(3),
        marginTop:'-30px'
    },
    dialogPaper:{
        maxWidth: '400px',
        maxHeight: '1800px',
        minHeight: '300px',
    },
}))

function Login(props){
    let history = useHistory()

    //Style
    const classes = useStyles()

    //State and eventHandler
    const[userName, setUserName] = useState('')
    const[password, setPassword] = useState('')


    const loginHandler = (e)=>{
        e.preventDefault()
        if(userName && password){
            axios.post("https://fakestoreapi.com/auth/login", {
                username : userName,
                password : password
            })
            .then((res)=>{
                console.log(res.data.token);
                if(res.data.token){
                    localStorage.setItem('token',res.data.token)
                    let activeUser = props.userDetail.filter(e => e.username===userName)
                    props.activeUser(activeUser)
                    props.loginActiveAction()
                    history.push('/dashboard')
                    props.loginErrorTypingHandler()
                }
                else{
                    props.loginErrorHandler()
                }
            })
            .catch((err)=>console.log(err))
        }
        else{
            props.loginErrorHandler()
        }
        
    }

    const err = props.loginError? 'error' : null

    const userNameHandler =(e)=>{
        setUserName(e.target.value)
        props.loginErrorTypingHandler()
    }
    
    const passwordHandler = (e)=>{
        setPassword(e.target.value)
        props.loginErrorTypingHandler()
    }

    return(
    <div data-testid = {props.dataTestid}>
        <Dialog fullWidth='true' maxWidth ='lg' open={props.loginActive} onClose={props.loginActiveAction} classes={{ paperWidthLg: classes.dialogPaper }}>
            <DialogTitle className={classes.title}>Login</DialogTitle>
                <form>
                    <DialogContent className={classes.content}>
                        <TextField
                            error={err}
                            label="User Name"
                            style={{ marginBottom: '20px' }}
                            fullWidth
                            size = 'medium'
                            margin="dense"
                            variant="outlined"
                            value = {userName}
                            onChange = {userNameHandler}
                            data-testid = 'login-username'
                        />
                        <TextField
                            error={err}
                            label="Password"
                            style={{ marginTop: 5 }}
                            fullWidth
                            type = 'password'
                            size = 'medium'
                            margin="dense"
                            variant="outlined"
                            value = {password}
                            onChange = {passwordHandler}
                        />
                        
                    </DialogContent>
                    <DialogActions>
                        <Button 
                            variant="contained" 
                            size="medium" 
                            color="primary" 
                            className={classes.margin}
                            onClick = {loginHandler}
                        >Login
                        </Button>
                    </DialogActions>
                </form>
        </Dialog>
    </div>
    )
}

const mapStateToProps = state =>{
    return{
        loginActive : state.loginActive,
        userDetail : state.userDetail,
        loginError : state.loginError
    }
}

const mapDispatchToProps = dispatch =>{
    return{
        loginActiveAction : ()=>dispatch(loginActive()),
        activeUser : (value)=> dispatch(activeUser(value)),
        loginErrorHandler : ()=>dispatch(loginError()),
        loginErrorTypingHandler : () =>dispatch(loginErrorTyping())
    }
}

export default connect(mapStateToProps,mapDispatchToProps)(Login)