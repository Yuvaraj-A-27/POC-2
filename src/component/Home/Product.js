import { Card, CardActionArea, CardContent, CardMedia, makeStyles, Typography } from '@material-ui/core';
import React from 'react';
import { useSelector } from 'react-redux';

const useStyles = makeStyles((theme)=>({
    heading:{
        textAlign:'left',
        marginTop:'3%',
        marginLeft : '3%',
        marginRight : '3%',
        fontWeight:'800',
        background: 'linear-gradient(45deg, #2c0ccc 10%, #7665c9 50%)',
        border: 0,
        fontSize: 20,
        borderRadius: 3,
        color: 'white',
        height: 25,
        padding: '10px 30px',
    },
    hr:{
        marginTop:'1%',
        marginLeft: theme.spacing(5),
        marginRight: theme.spacing(5),
        marginBottom: theme.spacing(4),

    },
    card:{
        maxWidth:'20%'
    },
    cardMedia:{
        height: '200px',
        width : '100%'
    }
}))

function Product(){
    const classes = useStyles()

    const product = useSelector(state=>state.product)
    let product1 = product[0]
    console.log(product);

    return(
        <div>
            <Typography className={classes.heading} variant ='h5'>Electronics</Typography>
            <hr className={classes.hr}/>
            <Card className ={classes.card}>
                <CardActionArea>
                    <CardMedia 
                        className={classes.cardMedia}
                        image={product1.image}
                        title = 'product1'
                    />
                    <CardContent>
                        <Typography gutterBottom variant="h5" component="h2">
                          Lizard
                        </Typography>
                        <Typography variant="body2" color="textSecondary" component="p">
                          Lizards are a widespread group of squamate reptiles, with over 6,000 species, ranging
                          across all continents except Antarctica
                        </Typography>
                    </CardContent>
                </CardActionArea>
            </Card>
        </div>
    )
}

export default Product;