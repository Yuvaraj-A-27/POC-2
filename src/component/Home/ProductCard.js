import { Card, CardActionArea, CardContent, CardMedia, makeStyles, Typography } from '@material-ui/core';
import React from 'react';

const useStyles = makeStyles(()=>({
    card:{
        marginTop:'50px',
        maxWidth:'80%',
        maxHeight:'100%',
        background: "linear-gradient(45deg, #a1bff0 1%, #cccccc 30%)",
        transition : 'transform 0.3s',
        '&:hover':{
            transform: 'translateY(10px)'
        }
    },
    cardMedia:{
        height: '200px',
        width : '100%',
    },
}))

function ProductCard({image, title, description}){
    const classes = useStyles()

    return(
            <Card className ={classes.card}>
                 <CardActionArea>
                     <CardMedia 
                         className={classes.cardMedia}
                         image={image}
                         title = {image}
                     />
                     <CardContent>
                         <Typography gutterBottom variant="h6" component="h2">
                           {title}
                         </Typography>
                     </CardContent>
                 </CardActionArea>
            </Card>
    )
}

export default ProductCard;