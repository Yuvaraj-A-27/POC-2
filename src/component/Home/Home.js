import axios from "axios";
import { useEffect } from "react";
import { connect } from "react-redux";
import { initialCategory, initialProduct, initialUserDetail} from "../../Store/Action";
import AppBarComponent from "./AppBarComponent";
import Category from "../util/Category";
import Product from "../util/Product";
import NavigationIcon from '@material-ui/icons/Navigation';
import { makeStyles } from "@material-ui/core";
import Login from "./Login";
import Register from "./Register";
import LeftNavBar from "../util/LeftNavBar";


const useStyles = makeStyles(()=>({
    upArrow:{
        position: 'sticky',
        background:'#2c0ccc' ,
        bottom: '7%',
        width: '80px',
        height: '60px',
        borderRadius:20,
        color:'white',
        marginLeft:'88%',
        transition : 'transform 0.2s',
        '&:hover':{
            transform: 'translateY(-10px)'
        }
    }
}))

function Home(props){

    const classes = useStyles()

    function datafetch(){
        axios.get("https://fakestoreapi.com/products/categories")
        .then((res) =>{
            props.initialCategory(res.data)
        })
        .catch((error) =>{
            console.log(error);
        })

        axios.get("https://fakestoreapi.com/products")
        .then((res) =>{
            props.initialProduct(res.data)
        })
        .catch((error) =>{
            console.log(error);
        })

        axios.get("https://fakestoreapi.com/users")
        .then((res) =>{
            props.initialUserDetail(res.data)
        })
        .catch((error) =>{
            console.log(error);
        })
    }
    
    useEffect(()=>{
        datafetch()
    },[])

    return(
        <div>
            <AppBarComponent dataTestid='AppBar' id='Appbar' />
            <Category dataTestid = 'Category' />
            <Product dataTestid = 'Product' />
            <a href='#Appbar'><NavigationIcon data-testid='navIcon' className={classes.upArrow} /></a>    
            <Login dataTestid='Login' />
            <Register dataTestid = 'Register' />
            <LeftNavBar />
        </div>
    )
}


const mapDispatchToProps = dispatch =>{
    return{
      initialCategory : (value) => dispatch(initialCategory(value)),
      initialProduct : (value) => dispatch(initialProduct(value)),
      initialUserDetail : (value) => dispatch(initialUserDetail(value))
    }
  }

export default connect(null,mapDispatchToProps)(Home)