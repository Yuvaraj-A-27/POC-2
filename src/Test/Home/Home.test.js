import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import Home from '../../component/Home/Home'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'


const middleware  = {}
const mockStore = configureStore(middleware)

test('Home Component Testing',()=>{

    const initialState = {}
    const Store = mockStore(initialState)

    const {getByTestId} = render(
        <Provider store={Store}>
            <Home />
        </Provider>)


    expect(getByTestId('AppBar')).toBeTruthy()
    expect(getByTestId('Category')).toBeTruthy()
    expect(getByTestId('Product')).toBeTruthy()
    expect(getByTestId('navIcon')).toBeTruthy()
    expect(getByTestId('Login')).toBeTruthy()



    //snapshot
    const home = renderer.create(
        <Provider store={Store}>
            <Home />
        </Provider>
    ).toJSON()

    expect(home).toMatchSnapshot()
})


