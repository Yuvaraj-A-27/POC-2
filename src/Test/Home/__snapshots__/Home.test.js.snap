// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Testing 1`] = `
<div>
  <div
    data-testid="AppBar"
  >
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary makeStyles-root-2 MuiPaper-elevation4"
      id="Appbar"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="menu"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-3 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <h5
          className="MuiTypography-root makeStyles-title-4 MuiTypography-h5"
        >
          MyStore
        </h5>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-btn-5 MuiButton-colorInherit"
          data-testid="register-btn"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Register
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-btn-5 MuiButton-colorInherit"
          data-testid="login-btn"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Login
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
  </div>
  <div
    data-testid="Category"
  >
    <h5
      className="MuiTypography-root makeStyles-heading-7 MuiTypography-h5"
    >
      Category
    </h5>
    <div
      className="makeStyles-paperDiv-9"
    />
    <hr
      className="makeStyles-hr-8"
    />
  </div>
  <div
    data-testid="Product"
  >
    <h5
      className="MuiTypography-root makeStyles-heading-12 MuiTypography-h5"
    >
      All Products
    </h5>
    <div
      className="sc-jSFjdj jvXOvp rec rec-carousel-wrapper "
      style={Object {}}
    >
      <div
        className="sc-eCApnc jCKguF rec rec-carousel"
        size={
          Object {
            "height": 0,
          }
        }
        style={
          Object {
            "height": 0,
          }
        }
      >
        <button
          className="sc-bdnxRM ehkxHv rec rec-arrow rec rec-arrow-left"
          disabled={true}
          onClick={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          ❮
        </button>
        <div
          className="sc-dlnjwi dTGAUq rec rec-slider-container"
        >
          <div
            className="sc-hKFxyN gXEQKd rec rec-slider"
            style={
              Object {
                "left": "0px",
                "right": "auto",
                "top": "auto",
                "transition": "all 500ms ease",
              }
            }
          >
            <div
              className="rec rec-swipable"
              onMouseDown={[Function]}
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            >
              <div
                className="rec rec-carousel-item rec-carousel-item-0 rec-carousel-item-visible"
              >
                <div
                  className="sc-gtsrHT cAlXfM rec rec-item-wrapper"
                  id={0}
                  onClick={[Function]}
                  style={
                    Object {
                      "padding": "0px 0px 0px 0px",
                      "width": "0px",
                    }
                  }
                >
                  <div
                    className="MuiPaper-root MuiCard-root makeStyles-card-14 MuiPaper-elevation1 MuiPaper-rounded"
                    data-testid="productCard"
                    onClick={[Function]}
                  >
                    <button
                      className="MuiButtonBase-root MuiCardActionArea-root"
                      disabled={false}
                      onBlur={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <div
                        className="MuiCardMedia-root makeStyles-cardMedia-15"
                      />
                      <div
                        className="MuiCardContent-root"
                      >
                        <h2
                          className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
                        />
                      </div>
                      <span
                        className="MuiCardActionArea-focusHighlight"
                      />
                      <span
                        className="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
              <div
                className="rec rec-carousel-item rec-carousel-item-1 rec-carousel-item-hidden rec-carousel-item-next"
              >
                <div
                  className="sc-gtsrHT cAlXfM rec rec-item-wrapper"
                  id={1}
                  onClick={[Function]}
                  style={
                    Object {
                      "padding": "0px 0px 0px 0px",
                      "width": "0px",
                    }
                  }
                >
                  <div
                    className="MuiPaper-root MuiCard-root makeStyles-card-14 MuiPaper-elevation1 MuiPaper-rounded"
                    data-testid="productCard"
                    onClick={[Function]}
                  >
                    <button
                      className="MuiButtonBase-root MuiCardActionArea-root"
                      disabled={false}
                      onBlur={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <div
                        className="MuiCardMedia-root makeStyles-cardMedia-15"
                      />
                      <div
                        className="MuiCardContent-root"
                      >
                        <h2
                          className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
                        />
                      </div>
                      <span
                        className="MuiCardActionArea-focusHighlight"
                      />
                      <span
                        className="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          className="sc-bdnxRM cGKbZh rec rec-arrow rec rec-arrow-right"
          disabled={false}
          onClick={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          ❯
        </button>
      </div>
      <div
        className="sc-iCoGMd xzcEF rec rec-pagination"
      >
        <button
          className="sc-gKAaRy ejXxNI rec rec-dot rec rec-dot_active"
          onClick={[Function]}
          tabIndex={-1}
          type="button"
        />
        <button
          className="sc-gKAaRy jLgnwX rec rec-dot "
          onClick={[Function]}
          tabIndex={0}
          type="button"
        />
      </div>
    </div>
    <hr
      className="makeStyles-hr-13"
    />
    <h5
      className="MuiTypography-root makeStyles-heading-12 MuiTypography-h5"
      id="electronics"
    >
      Electronics
    </h5>
    <div
      className="sc-jSFjdj jvXOvp rec rec-carousel-wrapper "
      style={Object {}}
    >
      <div
        className="sc-eCApnc jCKguF rec rec-carousel"
        size={
          Object {
            "height": 0,
          }
        }
        style={
          Object {
            "height": 0,
          }
        }
      >
        <button
          className="sc-bdnxRM ehkxHv rec rec-arrow rec rec-arrow-left"
          disabled={true}
          onClick={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          ❮
        </button>
        <div
          className="sc-dlnjwi dTGAUq rec rec-slider-container"
        >
          <div
            className="sc-hKFxyN gXEQKd rec rec-slider"
            style={
              Object {
                "left": "0px",
                "right": "auto",
                "top": "auto",
                "transition": "all 500ms ease",
              }
            }
          >
            <div
              className="rec rec-swipable"
              onMouseDown={[Function]}
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            />
          </div>
        </div>
        <button
          className="sc-bdnxRM ehkxHv rec rec-arrow rec rec-arrow-right"
          disabled={true}
          onClick={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          ❯
        </button>
      </div>
      <div
        className="sc-iCoGMd xzcEF rec rec-pagination"
      >
        <button
          className="sc-gKAaRy ejXxNI rec rec-dot rec rec-dot_active"
          onClick={[Function]}
          tabIndex={-1}
          type="button"
        />
      </div>
    </div>
    <hr
      className="makeStyles-hr-13"
    />
    <h5
      className="MuiTypography-root makeStyles-heading-12 MuiTypography-h5"
      id="jewelery"
    >
      Jewelery
    </h5>
    <div
      className="sc-jSFjdj jvXOvp rec rec-carousel-wrapper "
      style={Object {}}
    >
      <div
        className="sc-eCApnc jCKguF rec rec-carousel"
        size={
          Object {
            "height": 0,
          }
        }
        style={
          Object {
            "height": 0,
          }
        }
      >
        <button
          className="sc-bdnxRM ehkxHv rec rec-arrow rec rec-arrow-left"
          disabled={true}
          onClick={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          ❮
        </button>
        <div
          className="sc-dlnjwi dTGAUq rec rec-slider-container"
        >
          <div
            className="sc-hKFxyN gXEQKd rec rec-slider"
            style={
              Object {
                "left": "0px",
                "right": "auto",
                "top": "auto",
                "transition": "all 500ms ease",
              }
            }
          >
            <div
              className="rec rec-swipable"
              onMouseDown={[Function]}
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            />
          </div>
        </div>
        <button
          className="sc-bdnxRM ehkxHv rec rec-arrow rec rec-arrow-right"
          disabled={true}
          onClick={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          ❯
        </button>
      </div>
      <div
        className="sc-iCoGMd xzcEF rec rec-pagination"
      >
        <button
          className="sc-gKAaRy ejXxNI rec rec-dot rec rec-dot_active"
          onClick={[Function]}
          tabIndex={-1}
          type="button"
        />
      </div>
    </div>
    <hr
      className="makeStyles-hr-13"
    />
    <h5
      className="MuiTypography-root makeStyles-heading-12 MuiTypography-h5"
      id="men's clothing"
    >
      Men's Clothing
    </h5>
    <div
      className="sc-jSFjdj jvXOvp rec rec-carousel-wrapper "
      style={Object {}}
    >
      <div
        className="sc-eCApnc jCKguF rec rec-carousel"
        size={
          Object {
            "height": 0,
          }
        }
        style={
          Object {
            "height": 0,
          }
        }
      >
        <button
          className="sc-bdnxRM ehkxHv rec rec-arrow rec rec-arrow-left"
          disabled={true}
          onClick={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          ❮
        </button>
        <div
          className="sc-dlnjwi dTGAUq rec rec-slider-container"
        >
          <div
            className="sc-hKFxyN gXEQKd rec rec-slider"
            style={
              Object {
                "left": "0px",
                "right": "auto",
                "top": "auto",
                "transition": "all 500ms ease",
              }
            }
          >
            <div
              className="rec rec-swipable"
              onMouseDown={[Function]}
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            />
          </div>
        </div>
        <button
          className="sc-bdnxRM ehkxHv rec rec-arrow rec rec-arrow-right"
          disabled={true}
          onClick={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          ❯
        </button>
      </div>
      <div
        className="sc-iCoGMd xzcEF rec rec-pagination"
      >
        <button
          className="sc-gKAaRy ejXxNI rec rec-dot rec rec-dot_active"
          onClick={[Function]}
          tabIndex={-1}
          type="button"
        />
      </div>
    </div>
    <hr
      className="makeStyles-hr-13"
    />
    <h5
      className="MuiTypography-root makeStyles-heading-12 MuiTypography-h5"
      id="women's clothing"
    >
      Women's Clothing
    </h5>
    <div
      className="sc-jSFjdj jvXOvp rec rec-carousel-wrapper "
      style={Object {}}
    >
      <div
        className="sc-eCApnc jCKguF rec rec-carousel"
        size={
          Object {
            "height": 0,
          }
        }
        style={
          Object {
            "height": 0,
          }
        }
      >
        <button
          className="sc-bdnxRM ehkxHv rec rec-arrow rec rec-arrow-left"
          disabled={true}
          onClick={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          ❮
        </button>
        <div
          className="sc-dlnjwi dTGAUq rec rec-slider-container"
        >
          <div
            className="sc-hKFxyN gXEQKd rec rec-slider"
            style={
              Object {
                "left": "0px",
                "right": "auto",
                "top": "auto",
                "transition": "all 500ms ease",
              }
            }
          >
            <div
              className="rec rec-swipable"
              onMouseDown={[Function]}
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            />
          </div>
        </div>
        <button
          className="sc-bdnxRM ehkxHv rec rec-arrow rec rec-arrow-right"
          disabled={true}
          onClick={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          ❯
        </button>
      </div>
      <div
        className="sc-iCoGMd xzcEF rec rec-pagination"
      >
        <button
          className="sc-gKAaRy ejXxNI rec rec-dot rec rec-dot_active"
          onClick={[Function]}
          tabIndex={-1}
          type="button"
        />
      </div>
    </div>
    <hr
      className="makeStyles-hr-13"
    />
  </div>
  <a
    href="#Appbar"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root makeStyles-upArrow-1"
      data-testid="navIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2L4.5 20.29l.71.71L12 18l6.79 3 .71-.71z"
      />
    </svg>
  </a>
  <div
    data-testid="Login"
  />
  <div
    data-tesid="Register"
  />
</div>
`;
